test/like.test.js

```diff
@@ -0,0 +1,312 @@
+      .set('x-access-token', `${adminToken}`)
+      .end((error, res) => {
+        if (error) done(error);
+        expect(res.status).to.be.equal(401);
+        expect(res.body).to.have.deep.property('message');
+        expect(res.body.message).to.deep.equal('You can not dislike to your own post');
+        done();
+      });
+  });
+  it('should not unlike the article you didnt like', (done) => {
+    chai
+      .request(server)
+      .put('/api/v1/likes/unlike/fakeslug')
+      .set('Authorization', usertoken)
+      .end((error, res) => {
+        if (error) done(error);
+        expect(res).have.status(401);
+        expect(res).to.be.an('object');
+        expect(res.body).to.have.keys('message', 'status');
+        expect(res.body.message).to.deep.equal('you cant unlike article you did not like');
+        done();
+      });
+  });
+  it('Should successfully clap for a certain article', (done) => {
+    chai
+      .request(server)
+      .put('/api/v1/likes/clap/fakeslug')
+      .set('x-access-token', usertoken)
+      .end((error, res) => {
+        if (error) done(error);
+        expect(res).have.status(200);
+        expect(res).to.be.an('object');
+        expect(res.body).to.have.keys('message', 'status', 'data');
+        expect(res.body.message).to.deep.equal('Successfully claped');
+        expect(res.body.data).to.have.keys('ArticleSlug', 'status', 'claps');
+        expect(res.body.data.claps).to.deep.equal(1);
+        expect(res.body.data.status).to.deep.equal('like');
+        done();
+      });
+  });
+
+  it('Should successfully clap for a certain article second time', (done) => {
+    chai
+      .request(server)
+      .put('/api/v1/likes/clap/fakeslug')
+      .set('x-access-token', usertoken)
+      .end((error, res) => {
+        if (error) done(error);
+        expect(res).have.status(200);
+        expect(res).to.be.an('object');
+        expect(res.body).to.have.keys('message', 'status', 'data');
+        expect(res.body.message).to.deep.equal('Successfully claped');
+        expect(res.body.data).to.have.keys('ArticleSlug', 'status', 'claps');
+        expect(res.body.data.claps).to.deep.equal(2);
+        expect(res.body.data.status).to.deep.equal('like');
+        done();
+      });
+  });
+
+  it('should unlike Article', (done) => {
+    chai
+      .request(server)
+      .put('/api/v1/likes/unlike/fakeslug')
+      .set('Authorization', usertoken)
+      .end((error, res) => {
+        if (error) done(error);
+        expect(res).have.status(200);
+        expect(res).to.be.an('object');
+        expect(res.body).to.have.keys('message', 'status', 'data');
+        expect(res.body.message).to.deep.equal('Successfully unliked');
+        expect(res.body.data).to.have.keys('ArticleSlug', 'status', 'claps');
+        expect(res.body.data.claps).to.deep.equal(0);
+        expect(res.body.data.status).to.deep.equal('neutral');
+        done();
+      });
+  });
+
+  it('should dislike Article', (done) => {
```

Comment:
The test before this says 'should unlike Article' and this one says 'should dislike Article'. What is the difference between these two tests, and why do we have both of them?

Reply:
Unliking article is setting like to neutral, dislike is setting like status to dislike .
user can either clap , dislike and unlike an article but you can't unlike article which you didnt like or dislike before. hope thats is clear Noa .